BraveWallet/Crypto/Transactions/TransactionView.swift

```diff
@@ -0,0 +1,228 @@
+    $0.dateTimeStyle = .numeric
+  }
+  
+  private let numberFormatter = NumberFormatter().then {
+    $0.numberStyle = .currency
+    $0.currencyCode = "USD"
+  }
+  
+  private func namedAddress(for address: String) -> String {
+    NamedAddresses.name(for: address, accounts: keyringStore.keyring.accountInfos)
+  }
+  
+  private var gasFee: (String, fiat: String)? {
+    let isEIP1559Transaction = !info.txData.maxPriorityFeePerGas.isEmpty && !info.txData.maxFeePerGas.isEmpty
+    let limit = info.txData.baseData.gasLimit
+    let formatter = WeiFormatter(decimalFormatStyle: .gasFee(limit: limit.removingHexPrefix, radix: 16))
+    let gasFee = isEIP1559Transaction ? info.txData.maxFeePerGas : info.txData.baseData.gasPrice
+    if let value = formatter.decimalString(for: gasFee.removingHexPrefix, radix: .hex, decimals: Int(networkStore.selectedChain.decimals)) {
+      return (value, {
+        guard let doubleValue = Double(value), let assetRatio = assetRatios[networkStore.selectedChain.symbol.lowercased()] else {
+          return "$0.00"
+        }
+        return numberFormatter.string(from: NSNumber(value: doubleValue * assetRatio)) ?? "$0.00"
+      }())
+    }
+    return nil
+  }
+  
+  @ViewBuilder private var title: some View {
+    let formatter = WeiFormatter(decimalFormatStyle: .balance)
+    switch info.txType {
+    case .erc20Approve:
+      if info.txArgs.count > 1, let token = visibleTokens.first(where: {
+        $0.contractAddress == info.txArgs[0]
+      }) {
+        Text(String.localizedStringWithFormat(Strings.Wallet.transactionApproveSymbolTitle, formatter.decimalString(for: info.txArgs[1].removingHexPrefix, radix: .hex, decimals: Int(token.decimals)) ?? "", token.symbol))
+      } else {
+        Text(Strings.Wallet.transactionUnknownApprovalTitle)
+      }
+    case .ethSend, .other:
+      let amount = formatter.decimalString(for: info.txData.baseData.value.removingHexPrefix, radix: .hex, decimals: Int(networkStore.selectedChain.decimals)) ?? ""
+      let fiat = numberFormatter.string(from: NSNumber(value: assetRatios[networkStore.selectedChain.symbol.lowercased(), default: 0] * (Double(amount) ?? 0))) ?? "$0.00"
+      if info.isSwap {
+        Text(String.localizedStringWithFormat(Strings.Wallet.transactionSwapTitle, amount, networkStore.selectedChain.symbol, fiat))
+      } else {
+        Text(String.localizedStringWithFormat(Strings.Wallet.transactionSendTitle, amount, networkStore.selectedChain.symbol, fiat))
+      }
+    case .erc20Transfer:
+      if let token = visibleTokens.first(where: {
+        $0.contractAddress.caseInsensitiveCompare(info.txData.baseData.to) == .orderedSame
+      }) {
+        let amount = formatter.decimalString(for: info.txData.baseData.value.removingHexPrefix, radix: .hex, decimals: Int(token.decimals)) ?? ""
+        let fiat = numberFormatter.string(from: NSNumber(value: assetRatios[token.symbol.lowercased(), default: 0] * (Double(amount) ?? 0))) ?? "$0.00"
```

Comment:
will "$0.00" freaks out users? should we use something "n/a" to indicate that is some data/loading problem?

Reply:
Hmm, yeah this screen currently doesn't have any loaders on the transactions, but by the time you can view these screens chances are you have already loaded asset ratios for the assets shown since we only show the ones that are part of your visible portfolio assets.